### PR TITLE
Make: organize makefile defs

### PR DESCRIPTION
**Pull request by ldelossa**
house keeping task of organizing our Makefiles a bit neater.

this commit is step one a series of follow up commits for making our Makefiles more readable.

in this commit, various variables and helper functions are organized and segmented into section, hopefully making finding things a bit easier.

Signed-off-by: Louis DeLosSantos <louis.delos@isovalent.com>

```release-note
Segment Makefile defs into a series of sections.
```

### CODE REVIEW

**Comment by ldelossa** [comment]
In the process of testing all make targets available in `make help`

**Comment by ldelossa** [comment]
/test-1.26-net-next

File: Makefile [hunk 1]
```diff
@@ -113,6 +103,7 @@ define generate_k8s_protobuf
 endef
 
 build: check-sources $(SUBDIRS) ## Builds all the components for Cilium by executing make in the respective sub directories.
+	echo $(GO_VERSION_FILE)
```
Comment (joestringer) [inline Makefile]: In general I use `make --quiet V=0` specifically to get the minimal amount of output, and in such cases I prefer not to see spurious messages in the output that don't relate to the current step in the build process.

I suggest dropping it:

```suggestion
```

Another alternative is to preface the line with `$(QUIET)`, which will hide the output in the silent mode like how I run it above.

File: Makefile [hunk 2]
```diff
@@ -620,6 +611,7 @@ licenses-all: ## Generate file with all the License from dependencies.
 	@$(GO) run ./tools/licensegen > LICENSE.all || ( rm -f LICENSE.all ; false )
 
 check-go-version: ## Check locally install Go version against required Go version.
+	echo $(ROOT_DIR)
```
Comment (joestringer) [inline Makefile:614]:
```suggestion
```

**Review by joestringer — COMMENTED** [review]
First couple of things I noticed. I didn't look super closely at the rest given the number of lines changing, but I plan to take another look. I presume it's mostly just shuffling code from one place to another anyway.

File: Makefile (hunk 1 above)
Comment (ldelossa) [reply Makefile]: Okay, lets use $(QUIET) so that if someone does want to show the out, they can grep thru and remove that line.

**Pull request closed without merging**